### PR TITLE
disallow warnings for #if with -Wundef

### PR DESCRIPTION
**Pull request by tridge**
removed the whitelist of -Wundef for cpp files

### CODE REVIEW

File: ArduPlane/Attitude.cpp [hunk 1]
```diff
@@ -501,7 +501,7 @@ void Plane::stabilize()
             plane.stabilize_pitch(speed_scaler);
         }
 #endif
-#if ENABLE_SCRIPTING
+#ifdef ENABLE_SCRIPTING
```
Comment (peterbarker) [inline ArduPlane/Attitude.cpp:504]: Could we set `ENABLE_SCRIPTING` to zero in waf instead?

**Review by peterbarker — COMMENTED** [review]
https://github.com/ArduPilot/ardupilot/pull/19255

File: ArduPlane/Attitude.cpp (hunk 1 above)
Comment (tridge) [reply ArduPlane/Attitude.cpp:504]: only if you also change the 62 instances of "#ifdef ENABLE_SCRIPTING" to "#if"

**Comment by tridge** [comment]
replaced by #19255

**Pull request closed without merging**